packages/react-components/lib/tasks/create-task.tsx

```diff
@@ -803,6 +812,56 @@ export function CreateTaskForm({
     });
   };
 
+  enum DispatchType {
+    Automatic = 'Automatic',
+    Fleet = 'Fleet',
+    Robot = 'Robot',
+  }
+
+  const [dispatchType, setDispatchType] = React.useState<DispatchType>(DispatchType.Automatic);
+  const [robotDispatchTarget, setRobotDispatchTarget] = React.useState<RobotDispatchTarget | null>(
+    null,
+  );
+
+  const handleChangeDispatchType = (ev: React.ChangeEvent<HTMLInputElement>) => {
+    setDispatchType(ev.target.value as DispatchType);
+    setTaskRequest((prev) => {
+      return {
+        ...prev,
+        fleet_name: undefined,
+      };
+    });
+    setRobotDispatchTarget(null);
+  };
+
+  const handleDispatchFleetTargetChange = (ev: React.ChangeEvent<HTMLInputElement>) => {
+    setTaskRequest((prev) => {
+      return {
+        ...prev,
+        fleet_name: ev.target.value.length > 0 ? ev.target.value : undefined,
+      };
+    });
+  };
+
+  const handleDispatchRobotTargetChange = (ev: React.ChangeEvent<HTMLInputElement>) => {
+    if (!fleets) {
+      setRobotDispatchTarget(null);
+      return;
+    }
+    let robotFleet: string | null = null;
+    for (const fleetName of Object.keys(fleets)) {
+      if (fleets[fleetName].includes(ev.target.value)) {
+        robotFleet = fleetName;
+        break;
+      }
+    }
+    if (robotFleet === null) {
+      console.error(`Failed to find fleet name for robot [${ev.target.value}]`);
```

Comment:
The text field is not an auto-complete, therefore users can only ever select what is available, I added a comment to indicate in that section of code that it is highly unlikely this will happen

https://github.com/open-rmf/rmf-web/pull/1004/commits/35f92a8f12b85096b1f30f19674cabd396413434